File: Programmers/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    
    int n = s.size();
    map<char, int> m;
    
    for (int i = 0; i < n; i++) {
        char c = s[i];
            
        if (m.find(c) != m.end()) {
            answer.push_back(i - m[c]);
        } else {
            answer.push_back(-1);
        }
            
        m[c] = i;
    }
    
    return answer;
}

File: Programmers/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int n = s.length();
        
        int[] answer = new int[n];
        
        Map<Character, Integer> map = new HashMap<>();
        
        for (int i = 0; i < n; i++) {
            char c = s.charAt(i);
            
            if (map.containsKey(c)) {
                answer[i] = i - map.get(c);
            } else {
                answer[i] = -1;
            }
            
            map.put(c, i);
        }
        
        return answer;
    }
}